Create cloud-init-cloud-on-hypriot-x64.md

renevo opened the pull request:
By request of @DieterReuter 

Let me know if you want any changes, I will probably read it one or two more times myself.

renevo committed:
Create cloud-init-cloud-on-hypriot-x64.md

renevo committed:
Bad grammer is bad

renevo committed:
Adding some important links

renevo commented:
All done with my own edits. Let me know!

StefanScherer commented:
Awesome! Unfortunately our Circle CI integration didn't start. Running with hugo locally I got and error.
Can you please check?

```
$ docker run --rm -v $(pwd):/site -w /site -p 1313:1313 giantswarm/hugo
Unable to find image 'giantswarm/hugo:latest' locally
latest: Pulling from giantswarm/hugo
a671d06fb2e2: Pull complete 
a3ed95caeb02: Pull complete 
218d2204eeb5: Pull complete 
Digest: sha256:7a6fcfc938988945c9cd5c1999523df5856652f88cd628bab3059f3c0b49379f
Status: Downloaded newer image for giantswarm/hugo:latest
ERROR: 2017/10/29 Error parsing page meta data for post/cloud-init-cloud-on-hypriot-x64.md
ERROR: 2017/10/29 Near line 3 (last key parsed 'Tags'): Expected an array value terminator ',' or an array terminator ']', but got '"' instead.
ERROR: 2017/10/29 Near line 3 (last key parsed 'Tags'): Expected an array value terminator ',' or an array terminator ']', but got '"' instead.
CRITICAL: 2017/10/29 Errors reading pages: Error:Near line 3 (last key parsed 'Tags'): Expected an array value terminator ',' or an array terminator ']', but got '"' instead. for cloud-init-cloud-on-hypriot-x64.md
```

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
++++
+Categories = ["Docker", "Raspberry Pi", "ARM", "ARM64"]
+Tags = ["Docker", "Raspberry Pi", "ARM", "ARM64", "External" "cloud-init", "rpi-64"]
```
Here is a comma missing, which causes a build error!
Just change `"External"` into `"External",` and it will works

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+> Things may come to those who wait, but only the things left by those who hustle.
+
+Over the last year, a lot has happened in the Raspberry Pi and Docker communities, there are captains helping lead the charge, one of those, [Dieter Reuter](https://twitter.com/Quintus23M) really has been pushing the cause for ARM64 support with both Raspberry Pi 3 and LinuxKit. He isn't a single man army, the rest of the [Docker Pirates](http://blog.hypriot.com/crew/) over at [Hypriot](http://bl ...
```
s/captain/Docker Captains/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+Over the last year, a lot has happened in the Raspberry Pi and Docker communities, there are captains helping lead the charge, one of those, [Dieter Reuter](https://twitter.com/Quintus23M) really has been pushing the cause for ARM64 support with both Raspberry Pi 3 and LinuxKit. He isn't a single man army, the rest of the [Docker Pirates](http://blog.hypriot.com/crew/) over at [Hypriot](http://bl ...
+
+Building on the backs of these outstanding community members, I was finally able to spin up a RaspberryPi, exactly like I do in the "real world", just turn it on, let it self-configure, then software just runs.
```
s/RaspberryPi/Raspberry Pi/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+Before we talk about the "new" pirate way, let't talk about the "old" non-pirate way.
+
+* Download Jessie Lite (the easiest for nubbies like me)
```
s/Jessie/Raspbian/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+* Download Jessie Lite (the easiest for nubbies like me)
+* Flash said image to an SD Card
+* Optionally re-mount SD Card and monkey some stuff around
+* Put the card in your PI
```
s/PI/RPi/
I think using the term `RPi` as the abbreviation for `Raspberry Pi` seems a little bit more consistent

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+* Flash said image to an SD Card
+* Optionally re-mount SD Card and monkey some stuff around
+* Put the card in your PI
+* Power up the PI
```
s/PI/RPi/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+* Optionally re-mount SD Card and monkey some stuff around
+* Put the card in your PI
+* Power up the PI
+* If you don't have avahi/bonjour, go find the IP in your router
```
Tip: even without avahi/bonjour you can easily run a ping command on your network to detect the RPi: `ping <hostname>.local` or `ping <hostname>` (depends on your router capabilities).

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+* If you don't have avahi/bonjour, go find the IP in your router
+* SSH into the server
+* Update Packages
+* Run `raspi-config` and modify some junk.
```
Remove the trailing dot `.`

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+#### Resize File System
+
+This is something that had plagued RPI users for years, finally at some point the foundation images included this on first boot, but the wonderful thing about `cloud-init` is it already had this built in. You "really" don't need these settings in the file, I put these here to show that this is just built in functionality for `cloud-init`.
```
s/RPI/RPi/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+There are actually a few ways to run commands in `cloud-init`, this is the most used, because it is the last thing that is done during initialization.
+
+In this, we are essentially restarting docker and avahi to pickup our configuration changes, initializing docker swarm, running good ol Portainer (so we can see what is going on without SSH).
```
s/docker/Docker/
s/docker swarm/Docker Swarm/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+### Boot and Forget
+
+For this, I simply put the SD card into a random RPI3 I had laying around, plugged in a network cable to my switch and the PI, and then applied power to it.
```
s/RPI3/RPi3/
s/PI/RPi/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+For this, I simply put the SD card into a random RPI3 I had laying around, plugged in a network cable to my switch and the PI, and then applied power to it.
+
+This is really where the waiting game begins, the PI is going to boot up, self-name, self-update, possible reboot, pull down some docker images, bootstrap a swarm, run portainer, then finally... start a personal cloud.
```
s/PI/RPi/
s/docker images/Docker images/
s/swarm/Docker Swarm/
s/portainer/Portainer/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+This is really where the waiting game begins, the PI is going to boot up, self-name, self-update, possible reboot, pull down some docker images, bootstrap a swarm, run portainer, then finally... start a personal cloud.
+
+At the time of writing this, you are still going to have to go fish out your IP from DNS unless you are on Mac or Linux with avahi, at that case, you can just navigate to `http://<hostname>.local` or `http://<hostname>` if you used an fqdn like I did. There are actually ways to get the PI to phone home when it is finished bootstrapping, but I am going to save that for another post!
```
s/PI/RPi/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+At the time of writing this, you are still going to have to go fish out your IP from DNS unless you are on Mac or Linux with avahi, at that case, you can just navigate to `http://<hostname>.local` or `http://<hostname>` if you used an fqdn like I did. There are actually ways to get the PI to phone home when it is finished bootstrapping, but I am going to save that for another post!
+
+So, after you get some coffee, you can try navigating to your PI on port 9000 with your browser, that should get you into the Portainer instance without any type of authentication (don't do this in a real environment please, go read the docs on securing it).
```
s/PI/RPi/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+So, after you get some coffee, you can try navigating to your PI on port 9000 with your browser, that should get you into the Portainer instance without any type of authentication (don't do this in a real environment please, go read the docs on securing it).
+
+After you go get a snack and take a short walk, you can try navigating to your PI on port 80 with your browser, once you get prompted, login with user: `pirate` and password: `hypriot` to get access to your cloud. Click close on the annoying modal about downloading sync programs, and there you have it... Your own personal cloud, bootstrapped from a simple YAML file, without you ever having to SSH ...
```
s/PI/RPi/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+After you go get a snack and take a short walk, you can try navigating to your PI on port 80 with your browser, once you get prompted, login with user: `pirate` and password: `hypriot` to get access to your cloud. Click close on the annoying modal about downloading sync programs, and there you have it... Your own personal cloud, bootstrapped from a simple YAML file, without you ever having to SSH ...
+
+Here is the fun part, you can reflash that anytime you want to recreate the exact same baseline image.
```
s/image/SD card image/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+## Summary
+
+This was a lot of fun for me, and without having to actually figure out some nitty gritty details, not over documenting things, and getting prepared to write this post, it realistically took me about 10 minutes to go from downloading the OS image to running NextCloud on my PI. And that is all due to the hard work of the community, and especially the Docker Pirates at Hypriot, and to them, I thank ...
```
s/PI/RPi/

DieterReuter commented on content/post/cloud-init-cloud-on-hypriot-x64.md:
```diff
@@ -0,0 +1,348 @@
+
+This was a lot of fun for me, and without having to actually figure out some nitty gritty details, not over documenting things, and getting prepared to write this post, it realistically took me about 10 minutes to go from downloading the OS image to running NextCloud on my PI. And that is all due to the hard work of the community, and especially the Docker Pirates at Hypriot, and to them, I thank ...
+
+I plan to continue playing with this, potentially updating the server to use an external USB drive for data and auto-mounting it on boot, possible setup a Gluster FS and run them in a small 2 or 3 node cluster, and call it my PiCloud. 
```
s/Gluster FS/GlusterFS/

DieterReuter reviewed (commented):
BTW, what a great blog post!

renevo committed:
Feedback from PR

Fixed build error (missing comma)
Fixed consistencies in casing/naming of a few items.

StefanScherer commented:
LGTM. We thought about adding a nice image, eg. something like [this](https://www.flickr.com/photos/horiavarlan/5019447085/in/photolist-8DxYK8-6RFbEf-6RFfrw-6RLCGd-6RAH6n-6RF9gf-6RATnx-6RGCKM-6RB7L2-6REXqA-6T4VEQ-6RBeKg-brFbp6-6RAZU8-6RGwNk-6RAiLD-6RBgp6-6RLAvm-6RBbwM-6RAE5e-6RLJVj-6RLFdN-6RBsmr-6RFxxw-6RLJrY-6RFvGJ-6RFtZS-6REvrf-6RENUY-6RFwCC-6RGBEn-6RGz9n-6RFpb3-6RFzo1-if9tn-9NF66y-84UZy9-6RFew9-6RFq2m-6RLHWJ-6T4Wd9-9NChKc-6RAZ5V-9NCiK8-6T5Lbf-6RAUfp-6RBhb6-9NF3UW-9NF979-9NFaFW). :-)

renevo commented:
Feel free to add any image you like, I couldn't come up with one, except this one:

![cloud-init-rpi-cloud](https://user-images.githubusercontent.com/1626559/32147033-712a0d96-bc9d-11e7-8974-d8f0c4e17f5b.png)

DieterReuter commented:
LGTM
